server/services/emailNotifier.js

```diff
@@ -0,0 +1,131 @@
+import nodemailer from "nodemailer";
+import Mailgen from "mailgen";
+import dotenv from "dotenv";
+import User from "../models/User.js";
+import Chore from "../models/Chore.js";
+
+dotenv.config();
+
+export const startEmailService = async () => {
+  const today = new Date(); // get today's date
+
+  let config = {
+    service: "gmail",
+    auth: {
+      user: process.env.EMAIL,
+      pass: process.env.EMAIL_PASSWORD,
+    },
+  };
+
+  let transporter = nodemailer.createTransport(config);
+  let MailGenerator = new Mailgen({
+    theme: "cerberus",
+    product: {
+      name: "Chorekeeper",
+      link: process.env.FRONTEND, //add frontend link
+    },
+  });
+
+  try {
+    //get user
+    let users = await User.find({});
+    // let users = [{ email: "altosintiams@gmail.com" }];
+
+    // for loop to loop through all the users
+    for (let i = 0; i < users.length; i++) {
+      let user = users[i];
+      //collect user chores
+      let chores = await Chore.find({ _id: { $in: user.chores } });
+
+      //select todays chores
+      let tempChores = chores.filter((chore) => {
+        const nextOccurrence = new Date(chore.nextOccurrence);
+        return (
+          nextOccurrence.getFullYear() === today.getFullYear() &&
+          nextOccurrence.getMonth() === today.getMonth() &&
+          nextOccurrence.getDate() === today.getDate()
+        );
+      });
+      let todaysChores = tempChores.map((chore) => {
+        return {
+          Chore: chore.name,
+          Location: chore.location,
+          "Duration(minutes)": chore.duration / 60,
+          // Preference: chore.preference,
+        };
+      });
+
+      //select overdue chores
+      let overChores = chores.filter((chore) => {
+        const nextOccurrence = new Date(chore.nextOccurrence);
+        return (
+          nextOccurrence.getFullYear() < today.getFullYear() &&
```

Comment:
Since nextOccurrence is in milliseconds wouldn't direct comparison resolve to false if it is not done at the exact millisecond the nextOccurence is scheduled for

Reply:
I typically would just do date.setHours(0, 0, 0, 0); so that the comparisons happen at the same time on the day, but the change you made also works.